lib/elixir/src/elixir_exp_clauses.erl

```diff
@@ -18,6 +18,10 @@ def(Fun, Args, Guards, Body, E) ->
 
 clause(Meta, Kind, Fun, {'->', ClauseMeta, [_, _]} = Clause, E) when is_function(Fun, 3) ->
   clause(Meta, Kind, fun(X, Acc) -> Fun(ClauseMeta, X, Acc) end, Clause, E);
+clause(Meta, 'cond', _Fun, {'->', _, [[{'_', _, Atom}], _]}, E) when is_atom(Atom) ->
```

Comment:
Yeah, I agree, but couldn't think of a place to put this instead, at least without traversing the clauses prematurely in places like `elixir_exp_clauses:'cond'/3`. Suggestions?